src/basic/terminal-util.c

```diff
@@ -455,7 +455,7 @@ int acquire_terminal(
                                         goto fail;
                                 }
 
-                                r = fd_wait_for_event(fd, POLLIN, ts + timeout - n);
+                                r = fd_wait_for_event(notify, POLLIN, ts + timeout - n);
```

Comment:
adding an explicit lockfile is nasty. first of all, it requires everybody to actually use it, and that's not the case for ttys more often than not. Never forget that there are many consumers of tty devices not just us. I also dislike lock files a lot since they have no clear lifecycle: if the hw that they are supposed to protect is removed, the lock file just stays around and is never cleaned up anymore. In this day and age we shouldn't do something like that anymore